lib/Rex/Interface/Exec/Local.pm

```diff
@@ -119,6 +119,11 @@ sub exec {
 
   }
 
+  # we need to bitshift $? so that $? contains the right (and for all
+  # connection methods the same) exit code after a run()/i_run() call.
+  # this is for the user, so that he can query $? in his task.
+  $? >>= 8;
```

Comment:
Well this has historical reasons. The first connection method was Net::SSH2 which provided already bitshifted return values (afaik, which come directly from libssh2 library). So for all other connection methods i added this.

Also the normal shell behaviour (which admin knows) is to have "normal" return values and no need to bitshift them.